File: hmt1developerexamples/src/main/java/com/realwear/hmt1developerexamples/MainMenuTile.java
/**
 * RealWear Development Software, Source Code and Object Code
 * (c) RealWear, Inc. All rights reserved.
 * <p>
 * Contact info@realwear.com for further information about the use of this code.
 */

package com.realwear.hmt1developerexamples;

import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

/**
 * Class which represents a clickable time to start an example
 */
public class MainMenuTile extends LinearLayout implements View.OnClickListener {
    private String mLaunchClass;

    /**
     * Constructor
     *
     * @param context      The context
     * @param commandResID The speech command to activate the tile
     * @param imageResID   The image to display on the tile
     * @param launchClass  The class to launch when the tile is clicked by the user
     */
    public MainMenuTile(Context context, int commandResID, int imageResID, String launchClass) {
        super(context);

        mLaunchClass = launchClass;

        LayoutInflater inflater =
                (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        inflater.inflate(R.layout.command_square, this, true);

        setImage(imageResID);
        setSmallText(commandResID);
        setOnClickListener(this);
    }

    /**
     * Set the image displayed on the tile
     *
     * @param resID The image to display
     */
    public void setImage(int resID) {
        ImageView iv = (ImageView) findViewById(R.id.imageView);
        Drawable drawable = getResources().getDrawable(resID, getContext().getTheme());
        iv.setImageDrawable(drawable);
    }

    /**
     * Set the text displayed on the tile
     *
     * @param resID The text to display
     */
    public void setSmallText(int resID) {
        TextView tv = (TextView) findViewById(R.id.smallTextView);
        String text = getResources().getString(resID);
        tv.setText(text);
    }

    /**
     * Called when the tile is clicked
     *
     * @param view This tile
     */
    @Override
    public void onClick(View view) {
        String packageName = getContext().getPackageName();

        Intent intent = new Intent();
        intent.setComponent(new ComponentName(packageName, packageName + "." + mLaunchClass));
        getContext().startActivity(intent);
    }
}

File: hmt1developerexamples/src/main/java/com/realwear/hmt1developerexamples/MainMenuTile.kt
/**
 * RealWear Development Software, Source Code and Object Code
 * (c) RealWear, Inc. All rights reserved.
 *
 *
 * Contact info@realwear.com for further information about the use of this code.
 */
package com.realwear.hmt1developerexamples

import android.annotation.SuppressLint
import android.widget.LinearLayout
import android.content.Intent
import android.content.ComponentName
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import androidx.core.content.res.ResourcesCompat
import com.realwear.hmt1developerexamples.databinding.CommandSquareBinding

/**
 * Given the [context], MainMenuTile represents a clickable tile constructed with the icon
 * [commandResID] and title [imageResID].
 *
 * Launches [launchClass] when clicked.
 */
@SuppressLint("ViewConstructor")
class MainMenuTile (
    context: Context,
    private val commandResID: Int,
    private val imageResID: Int,
    private val launchClass: String
) : LinearLayout(context), View.OnClickListener {
    private val binding: CommandSquareBinding

    /**
     * Set the image displayed on the tile via [resID]
     */
    private fun setImage(resID: Int) {
        val drawable = ResourcesCompat.getDrawable(resources, resID, context.theme)
        binding.imageView.setImageDrawable(drawable)
    }

    /**
     * Set the text displayed on the tile via [resID]
     */
    private fun setSmallText(resID: Int) {
        binding.smallTextView.text = resources.getString(resID)
    }

    /**
     * Called when the [tile] is clicked
     */
    override fun onClick(tile: View) {
        val packageName = context.packageName
        val intent = Intent().apply {
            component = ComponentName(packageName, "$packageName.$launchClass")
        }
        context.startActivity(intent)
    }

    init {
        val layoutInflater =
            context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        binding = CommandSquareBinding.inflate(layoutInflater)
        setImage(imageResID)
        setSmallText(commandResID)
        setOnClickListener(this)
    }
}